libs/designer-ui/src/lib/createConnection/index.tsx

```diff
@@ -103,35 +103,71 @@ export const CreateConnection = (props: CreateConnectionProps): JSX.Element => {
     [selectedParamSetIndex]
   );
 
+  const singleAuthParams = useMemo(() => connectionParameters ?? {}, [connectionParameters]);
+  const multiAuthParams = useMemo(
+    () => connectionParameterSets?.values[selectedParamSetIndex].parameters ?? {},
+    [connectionParameterSets, selectedParamSetIndex]
+  );
+  const isMultiAuth = useMemo(() => (connectionParameterSets?.values?.length ?? 0) > 1, [connectionParameterSets?.values]);
+
+  const isOnlyGatewayCapable = useCallback(
+    (capabilities: string[] = []) =>
+      (capabilities?.includes(Capabilities[Capabilities.gateway]) && !capabilities?.includes(Capabilities[Capabilities.cloud])) ?? false,
+    []
+  );
+
+  const hasOnlyOnPremGateway = useMemo(() => isOnlyGatewayCapable(connectorCapabilities), [connectorCapabilities, isOnlyGatewayCapable]);
+
+  const showGatewayOption = !hasOnlyOnPremGateway;
+  const [optionalGatewayEnabled, setOptionalGatewayEnabled] = useState<boolean>(false);
+
+  const isOptionalGatewayParam = useCallback(
+    (parameter: ParamType) => {
+      const capabilities = parameter.uiDefinition?.constraints?.capability;
+      return isOnlyGatewayCapable(capabilities);
+    },
+    [isOnlyGatewayCapable]
+  );
+
   const isParamVisible = useCallback(
     (parameter: ParamType) => {
       const constraints = parameter?.uiDefinition?.constraints;
       if (constraints?.hidden === 'true' || constraints?.hideInUI === 'true') return false;
       const dependencyParam = constraints?.dependentParameter;
       if (dependencyParam && parameterValues[dependencyParam.parameter] !== dependencyParam.value) return false;
       if (parameter.type === ConnectionParameterTypes[ConnectionParameterTypes.oauthSetting]) return false;
+      if (!optionalGatewayEnabled && isOptionalGatewayParam(parameter)) return false;
```

Comment:
1. What is the scenario we are trying to achieve? I'm curious weather this feature is needed, because seems it add unnecessary complexity and doesn't align with how capabilities shall be used originally.
2. This code adds ad hoc logic that couples capabilities with visibility and type.

Reply:
I think what tony means here is, instead of having it as gateway option or gateway parameter, it should be a capability option or capability parameter, because that is how it was modeled i think.. capabilities are array of string, it can have cloud, gateway.. the previous code kept it very hard coded logic as to what capability we are looking for

Reply:
maybe you can have this as a function and write the specific logic there, this can check for a capability to enable and if the parameter only belongs to that capability?